src/common.ts

```diff
@@ -96,14 +94,26 @@ export const nullable = <T>(processor: ValueProcessor<T>): ValueProcessor<T | nu
   },
 });
 
-export const nullOr = <T, O>(factory: ValueProcessorFactory<T, O>): ValueProcessorFactory<T | null, O> => (options) => nullable(factory(options));
+export const nullOr = <T>(processor: ValueProcessor<T>): ValueProcessor<T | null> => nullable(processor);
+
+export const nullableAs = <T>(processor: ValueProcessor<T>, options?: WithDefault<T | null>): ValueProcessor<Exclude<T, undefined> | null> => ({
+  process: (value: unknown): Result<Exclude<T, undefined> | null> => {
+    if (value === null) return { value: null };
+
+    const result = processor.process(value);
+    if (options?.default !== undefined && !isIssue(result) && result.value === undefined) return { value: options.default as Exclude<T, undefined> | null };
+    return result as Result<Exclude<T, undefined>>;
+  },
+});
+
+export const nullOrAs = <T>(processor: ValueProcessor<T>, options?: WithDefault<T | null>): ValueProcessor<Exclude<T, undefined> | null> => nullableAs(processor, options);
```

Comment:
Can we please call it `asNullOr`

Reply:
the name nullOrAs comes from that the value can be null or as matching type of the given valueProcessor, the default can be null or a string in case of a wrapped string value processor

Reply:
done